### PR TITLE
Implemented search field for PVR views. (#695)

### PR DESCRIPTION
**Pull request by Selaron**
This implements a multi-word search for PVR Recordings, PVR Channels and PVR Radiochannels.
On channel views, when typing multiple words, a channel is displayed if all words each match either the channel name or the title of the current broadcast (match mode: case insensitive contains).
On recordings view, a recording is displayed if all words each match either the recording title or the channel name it has been recorded on (match mode: case insensitive contains)

For `AbstractSearchableFragment` I mostly copied code from `AbstractCursorListFragment` and I'm not sure if all of this is at all reqiured. I'm not sure what savedSearchFilter is used for, what `onPause()` or `onResume()` does and if `AbstractSearchableFragment#onCreateView` is required at all.

### CODE REVIEW

File: app/src/main/java/org/xbmc/kore/ui/AbstractSearchableFragment.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+import de.greenrobot.event.EventBus;
+
+public abstract class AbstractSearchableFragment extends Fragment implements SearchView.OnQueryTextListener {
+
+    private String searchFilter = null;
+    private String savedSearchFilter;
+    private boolean supportsSearch;
+    private boolean isPaused;
+
+    private EventBus bus;
```
Comment (SyncedSynapse) [inline app/src/main/java/org/xbmc/kore/ui/AbstractSearchableFragment.java]: I don't think this is needed.

File: app/src/main/java/org/xbmc/kore/ui/AbstractSearchableFragment.java [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    private SearchView searchView;
+
+    private final String BUNDLE_KEY_SEARCH_QUERY = "search_query";
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View root = super.onCreateView(inflater, container, savedInstanceState);
+
+        bus = EventBus.getDefault();
```
Comment (SyncedSynapse) [inline app/src/main/java/org/xbmc/kore/ui/AbstractSearchableFragment.java]: I don't think this is needed.

File: app/src/main/java/org/xbmc/kore/ui/AbstractSearchableFragment.java [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+        View view = searchView.findViewById(R.id.search_close_btn);
+        if (view != null) {
+            view.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    EditText editText = (EditText) searchView.findViewById(R.id.search_src_text);
+                    editText.setText("");
+                    searchView.setQuery("", false);
+                    searchFilter = savedSearchFilter = "";
+                    restartLoader();
```
Comment (SyncedSynapse) [inline app/src/main/java/org/xbmc/kore/ui/AbstractSearchableFragment.java]: Call something else to this function, as it has nothing to do with Android's Loader - refreshList() for instance.

File: app/src/main/java/org/xbmc/kore/ui/AbstractSearchableFragment.java [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+                break;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+
+
+    @Override
+    public void onResume() {
+ //       bus.register(this);
```
Comment (SyncedSynapse) [inline app/src/main/java/org/xbmc/kore/ui/AbstractSearchableFragment.java]: You've already commented it, remove it completly. Same in `onPause`

**Comment by SyncedSynapse** [comment]
I made some comments that point out things that i think are dispensable. Can you please check them, remove them and see if everything works or if i missed something.

Regarding your comments, `onPause`, `onResume`, `onCreateView` are part Android's view lifecycle and are required. The stuff related to the `EventBus` is unnecessary, i believe.

The `savedSearch` can be tested like this: 
1) type something in the search filter, and check that it filters
2) exit kore (with the home button)
3) get back to kore and check if the filter is still on, and coherent with what's displayed on the search box.

**Comment by Selaron** [comment]
Thanks for your quick feedback. I applied your suggestions and it works.

Trying to test `savedSearchFilter` I set a breakpoint where the field is written but this never gets hit.
My emulator is a Pixel3 API 28.
Never the less pressing home and then resuming the app also keeps the search field input and item list state as is. Maybe recent android versions simply preserve the full app state?

**Comment by Selaron** [comment]
Need to fix one issue: on channel programme guide, the search has no effect. When inputting sth. and going back to channel list, Kore crashes. I'll notify when done.

**Comment by poisdeux** [comment]
> Thanks for your quick feedback. I applied your suggestions and it works.
> 
> Trying to test `savedSearchFilter` I set a breakpoint where the field is written but this never gets hit.
> My emulator is a Pixel3 API 28.
> Never the less pressing home and then resuming the app also keeps the search field input and item list state as is. Maybe recent android versions simply preserve the full app state?

Yes it does. But not when you change the device orientation from portrait to landscape or vice versa. In that case it has to recreate the whole view and we need to restore the state by saving the search filter first and get it back from the `bundle` in `onCreateView`.

File: app/src/main/java/org/xbmc/kore/ui/AbstractSearchableFragment.java [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+    /** {@inheritDoc} */
+    @Override
+    public boolean onQueryTextChange(String newText) {
+        if ((!searchView.hasFocus()) && TextUtils.isEmpty(newText)) {
+            //onQueryTextChange called as a result of manually expanding the searchView in setupSearchMenuItem(...)
+            return true;
+        }
+
+        /**
+         * When this fragment is paused, onQueryTextChange is called with an empty string.
```
Comment (poisdeux) [inline app/src/main/java/org/xbmc/kore/ui/AbstractSearchableFragment.java]: Please update this comment as it is referring to a `cursoradapter` that is not used here.
If I understand correctly I think the first two sentences will do just fine.

**Comment by Selaron** [comment]
I fixed the crash und made the search field work on CannelEPGlist either.

I also confirmed that `savedSearchFilter` is functional and actually used when device orientation is changed thanks to @poisdeux 's hint.

**Pull request merged**